### ISSUE TITLE
[Accessibility] Keyboard navigation breaks after updating series in heatmap

### ISSUE THREAD

**Issue by magdalena-gut** [highcharts/highcharts#16751]
#### Expected behaviour
Keyboard navigation should work correctly after updating the series' data.

#### Actual behaviour
After the points are updated, the order of keyboard navigation is changed.  

#### Demo and steps to reproduce
Initially, you can focus the first point and navigate with arrows to the next points.
But when the button is clicked (setData), the navigation order is changed.

https://jsfiddle.net/BlackLabel/jLxdg6h1/

#### Workaround
Set `updatePoints` parameter to `false` to replace points instead of updating them: https://jsfiddle.net/BlackLabel/6ypj8mbe/

#### Product version
Highcharts 9.3.2
